### PR TITLE
do not allow over drag

### PR DESCRIPTION
**Pull request by jakerichan**
Dragging the image feels sluggish for my application. Optional drag speed will help anybody customize this feature for their application.

### CODE REVIEW

**Comment by jakerichan** [comment]
Just realized this has an issue with portrait images, I'm testing a fix now.

**Comment by jakerichan** [comment]
@mosch sorry for not testing this better before making the pull request. I've now tested it with multiple image sizes.

**Comment by mosch** [comment]
@jakerichan thanks for the contrib. it seems like we could save some calculations by reusing your getBoundedX/getBoundedY functions in calculatePosition, do you agree on this?

File: index.js [hunk 1]
```diff
@@ -238,14 +238,37 @@
                 var yMove = yDiff < 0 ? 'down' : 'up';
                 var xMove = xDiff < 0 ? 'right' : 'left';
```
Comment (jakerichan) [inline index.js]: Maybe consider removing these at some point :point_up: Since they aren't doing anything

**Comment by jakerichan** [comment]
@mosch I spent some time trying to make that work and came to the conclusion that we would have to do those calculations anyway to position the image in the canvas.  And there's really no reason to run it through getBoundedX/Y at that point, since by the time it gets to calculatePosition it's already reflected in image.x/y

File: index.js (hunk 1 above)
Comment (mosch) [inline index.js]: jip. i'll do that after the merge.

**Comment by mosch** [comment]
i agree, thank you!

**Pull request merged**